src/libs/actions/Policy.js

```diff
@@ -218,29 +216,21 @@ function removeMembers(members, policyID) {
     if (members.length === 0) {
         return;
     }
-
-    const employeeListUpdate = {};
-    _.each(members, login => employeeListUpdate[login] = null);
-    Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY_MEMBER_LIST}${policyID}`, employeeListUpdate);
-
-    // Make the API call to remove a login from the policy
-    DeprecatedAPI.Policy_Employees_Remove({
+    const membersListKey = `${ONYXKEYS.COLLECTION.POLICY_MEMBER_LIST}${policyID}`;
+    const optimisticData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: membersListKey,
+        value: _.object(members, Array(members.length).fill({pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE})),
+    }];
+    const failureData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: membersListKey,
+        value: _.object(members, Array(members.length).fill({errors: {[DateUtils.getMicroseconds()]: Localize.translateLocal('workspace.people.error.genericRemove')}})),
```

Comment:
I have a question with this failureData. It looks good to me, but assume it fails when removing a user. Assume the  the failure happens in PHP/auth layer. Now your PHP code [here](https://github.com/Expensify/Web-Expensify/pull/34618/files#diff-31780863b86f2d567a1f2e3743719ba58bf64ed59edb6506e6d812ae897a58a4R839) will throw an ExpError with an `errorMessage`. 
Since the API is a failure, the failureData here will get triggered to insert all members including removed ones with the error. Additionally the workspace.people.error.genericRemove is also inserted. 
So if you removed 3 users, and the PHP layer failed on user 2.
1. Won't all 3 users get `workspace.people.error.genericRemove` added to them since the API is considered a failure? 
2. Won't user 2 show `workspace.people.error.genericRemove` and `errorMessage` i.e., two error messages which could be confusing?

Anyway my concern is mostly point 1, showing user1 and user3 back even though they were successfully removed, because push notification would run first to remove them and then failureData from the ExpError of user2 would add them back, right?

Let me know if that makes sense, if so I think maybe we should not have failureData over here.

Reply:
Also for any es device won't it be weird that the PHP layer is providing the error in English while the app adds `workspace.people.error.genericRemove` in Spanish?

Reply:
> Also for any es device won't it be weird that the PHP layer is providing the ExpError in English while the app adds workspace.people.error.genericRemove in Spanish? and then the app would display both.

We don't care. We are not supporting multiple languages.

As for the other comment, you are incorrect on point 1, here's a test I did by adding a throw before calling sharePolicy on a specific user (I had removed 2 users):
<img width="374" alt="image" src="https://user-images.githubusercontent.com/521248/191482029-3a8aa4ef-25c3-4010-8502-c8a901c0bc30.png">

As you can see, you are correct on point 2. I thought we would replace it, but that's not correct since we merge... this is a problem in many other APIs, since this is the pattern we are following in lots of them... I don't have a great solution for it right now, because:
- If we add the generic error, we get this behavior
- If we don't add the generic error, we risk not showing any error at all if what we got from the request was a failure with no data back

Any ideas? In any case, I don't think I would want to block on this, given this problem is endemic and must probably be fixed in many other places.